lib/kafka-consumer.js

```diff
@@ -158,12 +184,26 @@ Consumer.prototype.deserialize = function (type, message) {
 Consumer.prototype._transformAvro = function (data, encoding, callback) {
   const topicName = data.topic;
 
+  if (!this.sr.keySchemas[topicName] && data.key) {
+    log.warn('_transformAvro() :: Warning, consumer did not find topic on SR for key schema:',
+      topicName);
+
+    try {
+      data.parsedKey = data.key
+        ? JSON.parse(data.key.toString('utf-8'))
+        : null;
+    } catch(ex) {
+      log.warn('_transformAvro() :: Error parsing key:', data.key,
+        'Error:', ex);
+    }
+  }
```

Comment:
how you are doing this logic twice (to keys and message), you can create a function to get the parsed content

Reply:
@ricardohbin I would tend to agree but the logic for `key` is a bit different. If the `keySchema` does not exist and the `key` is different than `null`, then I simply `JSON.parse` it and move on, but for the `value`, it always assumes it is there, and if it doesn't find a schema it goes to the `callback`. In my opinion it wouldn't warrant a function for this because the behaviour for `keys` and `values` are specific to its own.

Reply:
ok!